src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -1529,7 +1529,7 @@ private bool TryGetNumber(ReadOnlySpan<byte> data, out int consumed)
             Debug.Assert(resultExponent == ConsumeNumberResult.OperationIncomplete);
 
             _bytePositionInLine += i;
-            ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedEndOfDigitNotFound, nextByte);
+            ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedEndOfDigitNotFound, data[i]);
```

Comment:
This is what caused the error message to report the wrong value (saying 'e' was invalid, when it should be 'b'):

> 0e+1b
> BytesConsumed: 0, CurrentDepth: 0, TokenStartIndex: 0, TokenType: None, Exception: 'e' is an invalid end of a number. Expected a delimiter. LineNumber: 0 | BytePositionInLine: 4.
> BytesConsumed: 3, CurrentDepth: 0, TokenStartIndex: 0, TokenType: None, Exception: 'b' is an invalid end of a number. Expected a delimiter. LineNumber: 0 | BytePositionInLine: 4.

This change is something we can easily port to 3.0 (low risk), but it also isn't too impactful to the end-user.